tests/storage-vm

```diff
@@ -484,6 +484,32 @@ for poolDriver in $poolDriverList; do
 
         lxc delete -f v1
 
+        echo "==> Checking setting instances.migration.stateful for VMs"
+        # Check that migration.stateful is set in the instance's expanded config to the value of `instances.migration.stateful`
+        # if no local instance config nor profile value of `migration.stateful` is set.
+
+        lxc config set instances.migration.stateful=true
+        lxc init "${TEST_IMG:-ubuntu-minimal-daily:22.04}" v1 --vm -s "${poolName}"
+
+        lxc query /1.0/instances/v1 | jq '.expanded_config["migration.stateful"]' | grep -q true
+        lxc delete v1 -f
+        lxc config set instances.migration.stateful=false
+        lxc init "${TEST_IMG:-ubuntu-minimal-daily:22.04}" v1 --vm -s "${poolName}"
+        lxc query /1.0/instances/v1 | jq '.expanded_config["migration.stateful"]' | grep -q false
+        lxc delete v1 -f
+
+        lxc config set instances.migration.stateful=true
+        lxc init "${TEST_IMG:-ubuntu-minimal-daily:22.04}" v1 --vm --config migration.stateful=false -s "${poolName}"
+        lxc query /1.0/instances/v1 | jq '.expanded_config["migration.stateful"]' | grep -q false # the instance local config should take precedence
+        lxc delete v1 -f
+
+        lxc config set instances.migration.stateful=false
```

Comment:
I'd probably `unset` here too.

Reply:
But I think I forgot to do a `lxc config unset instances.migration.stateful` at the end to avoid potential side effects on the following tests.